packages/charts/react-charts-preview/library/src/components/DonutChart/DonutChart.tsx

```diff
@@ -190,17 +190,11 @@ const DonutChartBase: React.FunctionComponent<DonutChartProps> = React.forwardRe
       );
     }
 
-    function _addDefaultColors(donutChartDataPoint?: ChartDataPoint[]): ChartDataPoint[] {
+    function _addDefaultGradients(donutChartDataPoint?: ChartDataPoint[]): ChartDataPoint[] {
       return donutChartDataPoint
         ? donutChartDataPoint.map((item, index) => {
-            let defaultColor: string;
-            if (typeof item.color === 'undefined') {
-              defaultColor = getNextColor(index, 0);
-            } else {
-              defaultColor = getColorFromToken(item.color);
-            }
-            return { ...item, defaultColor };
-          })
+          return { ...item, gradient: item.gradient ?? getNextGradient(index, 0) };
```

Comment:
what happens if the gradient colors are undefined

Reply:
if `item.gradient == undefined` it will use the default palette

But if `item.gradient == ["", ""]` the arc wont be visible. I can update it to also check for this condition

Reply:
Updated to check if both colors are defined, and use the default palette otherwise